source/extensions/filters/listener/tls_inspector/tls_inspector.cc

```diff
@@ -228,15 +228,25 @@ std::vector<absl::string_view> Filter::getAlpnProtocols(const unsigned char* dat
     return protocols;
   }
 
-  absl::string_view str(reinterpret_cast<const char*>(data));
-  for (unsigned int i = 0; i < len;) {
-    uint32_t protocol_length = 0;
-    protocol_length <<= 8;
-    protocol_length |= data[i];
-    ++i;
-    absl::string_view protocol(str.substr(i, protocol_length));
-    protocols.push_back(protocol);
-    i += protocol_length;
+  // The data buffer contains a sequence of non-terminated strings, where each
+  // string is preceded by 1 byte that gives it's length (excluding the length
+  // byte itself). No string can be zero length, and the last string must not
+  // overrun or underun the buffer.
```

Comment:
perfect, thank you for adding this. 
just one nit: s/underun/underrun/

Reply:
Done